ismatec/util.py

```diff
@@ -23,87 +23,79 @@ class Communicator(threading.Thread):
     It handles the communication via direct serial or through a serial
     server, and keeps track of the messy mix of synchronous (command)
     and asynchronous (status) communication.
+
+    This communicator uses a threaded queue to provide non-blocking
+    serial communication. TODO explore async serial development.
```

Comment:
The old communicator base class was leaky. The boolean storage, needing to re-implement the same loop in subclasses, and slightly different function names were making it not too helpful of a base class.

In this version, the base Communicator clearly serves one function (request/response queue) and the methods tell that story imo.

The subclasses then just need to tell the base class how to read and write.